py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py

```diff
@@ -204,240 +203,80 @@ def layer_norm(
     return layer_norm.get_output(0)
 
 
-def native_group_norm(
+def group_norm(
     ctx: ConversionContext,
     target: Target,
     source_ir: Optional[SourceIR],
     name: str,
     input: TRTTensor,
-    weight: Optional[Union[torch.Tensor, np.ndarray]],
-    bias: Optional[Union[torch.Tensor, np.ndarray]],
+    weight: Optional[Union[TRTTensor, torch.Tensor, np.ndarray]],
+    bias: Optional[Union[TRTTensor, torch.Tensor, np.ndarray]],
     N: int,
     C: int,
     HxW: int,
     group: int,
     eps: float,
-    return_mean_rstd: bool = True,
-) -> Union[TRTTensor, Sequence[TRTTensor]]:
-    # TODO: Ask TRT team about the usage of INormalization Layer usage with num_groups and update the implementation
-    #       with INormalization Layer
+) -> Tuple[TRTTensor, torch.Tensor, torch.Tensor]:
     assert (
         len(input.shape) >= 3
-    ), f"The input dimension should not be less than 3, got {len(input.shape)}!"
+    ), f"Expected at least 3 dimensions for input tensor but got {len(input.shape)}"
 
-    B = input.shape[0]
-    # if C is provided, it must be as same as the channel from the input shape,
-    # else if C is zero, we should get the channel from the input shape
-    if C == 0:
-        C = input.shape[1]
     assert (
         C == input.shape[1]
-    ), f"The number of Channel={C} must be equal to the number of channels in the input shape={input.shape[1]}"
```

Comment:
Should we keep a check for H*W too? Like we keep for C?

Reply:
`HxW` argument is not used in the converter so I'm not sure if there is any meaning for checking `HxW`? And the users of `nn.GroupNorm` do not have to specify this argument anyway.

Reply:
Pytorch has a assertion: `Expected X.numel() == N * C * HxW`. If we only check `C` but not `N` or `HxW`, which may cause results mismatch between TRT and pytorch, e.g.,
```
class TestNativeGroupNormConverter(DispatchTestCase):
    def test_groupnorm1d(self):
        class GroupNorm(torch.nn.Module):
            def forward(self, x):
                return torch.ops.aten.native_group_norm.default(
                    x,
                    torch.ones((6,)),
                    torch.zeros((6,)),
                    2,
                    6,
                    448,
                    2,
                    1e-05,
                )[0]

        inputs = [torch.randn(4, 6, 224)]
        self.run_test(
            GroupNorm(),
            inputs,
        )
```

I think we should keep the behavior with pytorch.

Reply:
I see your point. But I'd like to know how did you get that assertion? I tried the following code and PyTorch didn't throw any error.

```py
import torch


class MyModule(torch.nn.Module):
    def __init__(self) -> None:
        super().__init__()

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        return torch.ops.aten.native_group_norm.default(
            x, torch.ones(6, device="cuda"), torch.zeros(6, device="cuda"), 2, 6, 448, 2, 1e-05
        )[0]


model = MyModule().eval().cuda()
x = torch.randn(4, 6, 224, device="cuda")
output = model(x)
print(output)
print(f"{output.shape=}")
```

In reality, the `N, C, HxW` arguments are calculated and supplied by PyTorch for the users of `torch.nn.GroupNorm` depending on the shape of input tensor. It's only us that directly call the aten operator rather than using `torch.nn.GroupNorm` like normal users do. Things become more complicated with dynamic shapes if we would check `N` and `HxW` because the value of dynamic dim is `-1` and then a few extra steps need to be done.

Reply:
Ah. The key is to have unequal numel. In my code above the numel was the same (2 * 6 * 448 == 4  * 6 * 224 == 5376) and hence the error wasn't triggered.

Actually adding check for `N` and `HxW` is easy for static-shaped input. The real problem for me is dynamic-shaped input. Since the value is `-1` for dynamic dim, either `impl.shape.shape` or `impl.shape.get_shape_with_dynamic_shape` need to be invoked to get the real shape. But the returned value is of type `trt.ITensor` and I don't know how to compare it with scalar type.